src/app/shared/navbar/navbar.component.ts

```diff
@@ -18,26 +20,23 @@ export class NavbarComponent implements OnInit, OnDestroy {
     private categoriesService: CategoriesService,
     private authService: AuthService,
     private router: Router,
+    private recipesService: RecipesService,
   ) {}
   public userSubscription: Subscription;
 
   ngOnInit(): void {
-    this.userSubscription = this.authService.user.subscribe(
-      (user) => {
-        if (user.token) {
-          this.isLoggedIn = true;
-          console.log(this.isLoggedIn);
-        }
-        if (!user.token) {
-          this.isLoggedIn = false;
-          console.log('loggedOut');
-        }
-      },
-      //if user token expiration date valid then change is logged into true
-    );
+    this.userSubscription = this.authService.user.subscribe((user) => {
```

Comment:
```
 public userSubscription: Subscription = new Subscription();
```
```t
his.userSubscription.add(this.authService.user.subscribe(...))
```
```
ngOnDestroy(): void {
   this.userSubscription.unsubscribe()
}
```

Reply:
Also extract this.authService.user.subscribe... into separate method that returns subscription